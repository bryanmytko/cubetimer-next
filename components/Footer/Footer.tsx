import Link from "next/link";

const Footer = () => {
  return (
    <footer className="h-10 dark:bg-gray-900 sticky top-[100vh] px-2 sm:px-4">
      <p className="text-gray-500 text-xs text-center pt-3">
        &copy; 2014-{new Date().getFullYear()} &hearts;&nbsp;
        <Link href="http://github.com/bryanmytko">Bryan Mytko</Link>
      </p>
    </footer>
  );
};

export default Footer;
