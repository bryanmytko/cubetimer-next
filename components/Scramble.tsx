interface ScrambleProps {
  scramble: string;
}

const Scramble = (props: ScrambleProps) => (
  <p className="text-white text-2xl text-center">{props.scramble}</p>
);

export default Scramble;
