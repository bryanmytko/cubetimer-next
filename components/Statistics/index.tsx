export { default as Solves } from "./Solves";
export { default as ClassicSolves } from "./ClassicSolves";
