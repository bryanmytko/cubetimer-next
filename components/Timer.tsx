import React, { useEffect, useReducer } from "react";

import Scrambler from "../lib/scrambler";
import { TimerReducer, TimerActionKind } from "../reducers";
/* import { FormatService, ScrambleService } from '../../services'; */

//const { humanReadableTime } = FormatService;

interface TimerState {
  running: boolean;
  ready: boolean;
  time: number;
  solveTimes: Array<number>;
  scramble: string;
}

const initialState: TimerState = {
  running: false,
  ready: false,
  time: 0,
  solveTimes: [],
  scramble: new Scrambler("3x3").generate(),
};

const Timer = () => {
  const [state, dispatch] = useReducer(TimerReducer, initialState);

  useEffect(() => {
    let interval: NodeJS.Timeout;

    if (state.running) {
      interval = setInterval(
        () => dispatch({ type: TimerActionKind.TICK }),
        10
      );
    }

    const handleKeydown = (e: KeyboardEvent) => {
      if (e.key !== " ") return;
      e.preventDefault();
      dispatch({ type: TimerActionKind.READY });
    };

    const handleKeyup = (e: KeyboardEvent) => {
      if (e.key !== " ") return;
      e.preventDefault();
      dispatch({ type: TimerActionKind.TOGGLE });
    };

    window.addEventListener("keydown", handleKeydown);
    window.addEventListener("keyup", handleKeyup);

    return () => {
      clearInterval(interval);
      window.removeEventListener("keydown", handleKeydown);
      window.removeEventListener("keyup", handleKeyup);
    };
  }, [state.running]);

  return (
    <div>
      <div className="card rounded bg-gray-50 dark:bg-slate-700 py-4 mx-auto mt-6 text-center w-4/5">
        <span className="text-6xl text-white">{state.time}</span>
      </div>
      <button
        id="timer-btn"
        className={`timer-btn-start block mx-auto mt-6 px-20 py-5 rounded-md ${state.ready ? "dark:bg-red-500" : "dark:bg-green-500"
          }`}
        onClick={() => dispatch({ type: TimerActionKind.TOGGLE })}
      >
        Press spacebar or click to begin!
      </button>
    </div>
  );
};

export default Timer;
