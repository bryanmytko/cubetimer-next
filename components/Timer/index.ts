export { default as Clock } from "./Clock";
export { default as ClockButton } from "./ClockButton";
export { default as Panel } from "./Panel";
export { default as Timer } from "./Timer";
export { default as Times } from "./Times";
