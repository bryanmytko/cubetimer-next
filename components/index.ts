export { Footer } from './Footer';
export { Header } from './Header';
export { Layout } from './Layout';
export { Scramble } from './Scramble';
export { Timer } from './Timer';