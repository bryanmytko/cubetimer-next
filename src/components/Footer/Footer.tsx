import { CoffeeButton, CopyText, Feedback } from "./";

const Footer = () => {
  return (
    <footer className="lg:h-20 py-4 bg-black sticky top-[100vh] px-2 sm:px-4 flex flex-col lg:flex-row gap-4 lg:gap-8 justify-center">
      <CopyText />
      <p className="text-gray-500 text-s text-center pt-3">•</p>
      <Feedback />
      <p className="text-gray-500 text-s text-center pt-3">•</p>
      <CoffeeButton />
    </footer>
  );
};

export default Footer;
