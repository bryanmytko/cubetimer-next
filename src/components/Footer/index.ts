export { default as Footer } from "./Footer";
export { default as CoffeeButton } from "./CoffeeButton";
export { default as CopyText } from "./CopyText";
export { default as Feedback } from "./Feedback";
