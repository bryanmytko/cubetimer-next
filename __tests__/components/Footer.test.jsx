import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import "@testing-library/jest-dom";

import { Footer } from "../../components/Footer";

describe("<Footer>", () => {
  it("renders the component", () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });

  it("displays the correct copyright date", () => {
    render(<Footer />);
    const date = new Date().getFullYear();
    const re = new RegExp(date);

    expect(screen.getByText(re)).toBeInTheDocument();
  });
});
