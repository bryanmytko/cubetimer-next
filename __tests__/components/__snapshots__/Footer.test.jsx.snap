// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> renders the component 1`] = `
<div>
  <footer
    class="h-10 bg-gray-900 sticky top-[100vh] px-2 sm:px-4"
  >
    <p
      class="text-gray-500 text-xs text-center pt-3"
    >
      © 2014-
      2023
       ♥ 
      <a
        href="http://github.com/bryanmytko"
      >
        Bryan Mytko
      </a>
    </p>
  </footer>
</div>
`;
